src/main/java/com/allenai/ml/linalg/DenseVector.java

```diff
@@ -0,0 +1,64 @@
+package com.allenai.ml.linalg;
+
+import lombok.val;
+
+import java.util.Arrays;
+
+public class DenseVector implements Vector {
```

Comment:
This PR is primarily about having an interface for vectors I know I need for ML and basic implementations which really wrap around 3rd parties. Even if the implementation that ends up being best is someone else's I'd end up doing an adapter class. The reason for that is that I strongly believe you want your interfaces in terms of library internal classes or JDK class. 

So I'm sure if performance became an issue with these choices, I'd do an implementation of the interface with those libraries. For the specific case of 1D vectors, I don't think it really matters. The links you have are really for matrix operations.